### DISCUSSION TITLE
Query ~ use_sdl2_render_thread (amiberry.conf)

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1139] [Q&A] [answered]
In the wiki,

>Use a separate thread when rendering SDL2 textures. This might provide a performance boost, but it's not supported in all SDL2
> back-ends (e.g. works fine under X11 but fails on KMSDRM/console on RPI4).
>
>Possible values: yes or no
>
>Default value: no (or yes, depends on selected target)

Background: I was helping a user on EAB forum --- they had imported the amiberry v5.6.1 binary release, over top of the Pimiga3 ~/pi/amiberry directory (stated as being amiberry v5.5b), and since that time they had audio issues. The fix seems to be to set the option 'use_sdl2_render_thread=yes' in amiberry.conf (this is the default for the pimiga amiberry v5.5b setup), and the audio problems were fixed. I deduce what happened was the v5.6.1 amiberry binary release had replaced the existing conf/amiberry.conf file that shipped with Pimiga3 ... which would've set the option 'use_sdl2_render_thread=no' ..... which seems to be the default?

This makes me question the words in brackets "(or yes, depends on selected target)" .... for example, if I compile amiberry from source, on Debian x86-64 in X, do I expect use_sdl2_render_thread=yes because as stated, "works fine under X11" ? It isn't of course, the default amiberry.conf file defines use_sdl2_render_thread=no ...

Ergo, question: for which 'selected target' is the value set to 'use_sdl2_render_thread=yes', because AFAICT regardless of selected target, amiberry.conf defaults to  use_sdl2_render_thread=no?

TIA

**Comment by midwan** [comment, marked as answer]
Several targets have that option as default in the Makefile (look for `-DUSE_RENDER_THREAD`). When that is defined, the `use_sdl2_render_thread` option is enabled, however this does not override what you may already have in the `amiberry.conf` file.

So in other words, the default option applies if there is no `amiberry.conf` file yet, in which case it's regenerated with the default values. If there is a file already, the values in there are respected. 

The file included in the repo has that value disabled for maximum compatibility, so if you overwrite your local file with the one from the repo, you would end up disabling that option. I'm guessing that's what happened in the case you mention above.

**Comment by giantclambake** [comment]
Thanks, understood ~ that would indeed explain the case mentioned.
